### PR TITLE
added breadcrumb and loader

### PR DESCRIPTION
**Pull request by Sree0076**
added breadcrumb and button loader

### CODE REVIEW

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts [hunk 1]
```diff
@@ -3,29 +3,33 @@ import { ConfirmationService } from 'primeng/api';
...
-    this.confirmationService.confirm({
-     header: this.dialogHeader,
-     icon: this.dialogIcon,
-     message: this.message,
-     accept: () => {
-        this.onConfirm.emit();
-     }
-    });
+  rejectChange(): void {
+    console.log("reject")
```
Comment (athul-sanjeevan) [inline src/app/shared/confirm-dialog/confirm-dialog.component.ts]: remove console log

File: src/app/features/vendor/vendor-view/vendor-view.component.ts [hunk 2]
```diff
@@ -69,8 +72,8 @@ export class VendorViewComponent {
    * Method to show confirm pop up to approve the vendor.
    */
   confirmApproval(): void {
-    this.confirmationService.confirm({
-      message: 'Are you sure you want to approve this vendor?',
+    this.isButtonLoading = true;
+    this.confirmationService.confirm({ message: 'Are you sure you want to approve this vendor?',
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-view/vendor-view.component.ts]: keep the message in the config file

File: src/app/features/vendor/vendor-view/vendor-view.component.ts [hunk 3]
```diff
@@ -33,21 +34,23 @@ import { IVendor } from '../models/vendor.model';
   templateUrl: './vendor-view.component.html',
   styleUrl: './vendor-view.component.scss',
 })
-export class VendorViewComponent {
+export class VendorViewComponent implements OnInit {
   vendor!: IVendor;
-
+  isButtonLoading = false;
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-view/vendor-view.component.ts:40]: keep a line gap

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 4]
```diff
@@ -160,51 +167,71 @@ export class VendorCreationComponent implements OnInit {
...
+    this.isConfirmPopupVisible = false;
+    this.router.navigate([""]);
+  }
+
+  /**
+   * Method to go handle reject from confirmation popup.
+   */
+  handleRejection(): void {
+    this.isConfirmPopupVisible = false;
+    console.log(this.isConfirmPopupVisible);
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: remove console.log

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 5]
```diff
@@ -54,29 +55,33 @@ export class VendorCreationComponent implements OnInit {
...
   vendorData?: IVendorData;
-  countries: string[] = ['USA', 'Germany', 'Australia', 'Brazil', 'India'];
-  states: string[] = ['California','Berlin','Sydney','Rio de Janeiro','Kerala',];
+  countries: string[] = ["USA", "Germany", "Australia", "Brazil", "India"];
+  states: string[] = ["California", "Berlin", "Sydney", "Rio de Janeiro", "Kerala",];
   markets!: IDropDownFields[];
   services!: IDropDownFields[];
   selectedVendorId!: number;
-  confirmationMessage = '';
+  confirmationMessage = "Are you sure you want to cancel the changes?";
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: keep this message in the config file

File: src/app/shared/confirm-dialog/confirm-dialog.component.ts (hunk 1 above)
Comment (Sree0076) [reply src/app/shared/confirm-dialog/confirm-dialog.component.ts]: done

File: src/app/features/vendor/vendor-view/vendor-view.component.ts (hunk 2 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-view/vendor-view.component.ts]: Done

**Review by athul-sanjeevan — APPROVED** [review]

File: src/app/shared/loader/loader.component.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { Component, Input } from '@angular/core';
+
```
Comment (kavyamolr) [inline src/app/shared/loader/loader.component.ts:2]: LLD missing

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 7]
```diff
@@ -27,27 +28,31 @@ import { Router } from '@angular/router';
...
+    });
+  }
 
   /**
-   * Method to get the list of vendors 
+   * Method to get the list of vendors
    */
   navigateToView(event: IVendor): void {
-   this.router.navigate(['/vendor/view/'+event.id]);
+    this.router.navigate(["/vendor/view/" + event.id]);
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: use single quotes. update in every ts file.

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 4 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts (hunk 5 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: done

File: src/app/shared/loader/loader.component.ts (hunk 6 above)
Comment (Sree0076) [reply src/app/shared/loader/loader.component.ts:2]: done

File: src/app/features/vendor/config/vendor-config.ts [hunk 8]
```diff
@@ -1,2 +1,18 @@
-export const phonePattern = new RegExp(/^[1-9][0-9]{9}$/);
-export const urlPattern = new RegExp(/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}(\.[a-z]{2,6})\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/);
\ No newline at end of file
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root', 
+})
+export class Constants {
+    public phonePattern: RegExp;
```
Comment (kavyamolr) [inline src/app/features/vendor/config/vendor-config.ts]: every variable is public by default. No need to mention it as public.

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 7 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: ok

File: src/app/features/vendor/config/vendor-config.ts (hunk 8 above)
Comment (Sree0076) [reply src/app/features/vendor/config/vendor-config.ts]: ok

File: src/app/shared/ui/button-loader.directive.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import { Directive, ElementRef, Renderer2, Input } from '@angular/core';
+
```
Comment (kavyamolr) [inline src/app/shared/ui/button-loader.directive.ts:2]: Add LLD

Comment (Sree0076) [reply src/app/shared/ui/button-loader.directive.ts:2]: Added

File: src/app/features/vendor/vendor-view/vendor-view.component.ts (hunk 3 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-view/vendor-view.component.ts:40]: done

**Review by kavyamolr — APPROVED** [review]

**Pull request merged**